editor/plugins/tiles/tile_map_editor.cpp

```diff
@@ -321,6 +321,7 @@ void TileMapEditorTilesPlugin::_update_patterns_list() {
 	for (int i = 0; i < tile_set->get_patterns_count(); i++) {
 		int id = patterns_item_list->add_item("");
 		patterns_item_list->set_item_metadata(id, tile_set->get_pattern(i));
+		patterns_item_list->set_item_tooltip(id, vformat(TTR("ID: %d"), i));
```

Comment:
Note that patterns within the TileSet are referred by index, see: [TileSet.add_pattern](https://docs.godotengine.org/en/4.0/classes/class_tileset.html#class-tileset-method-add-pattern), [TileSet.get_pattern](https://docs.godotengine.org/en/4.0/classes/class_tileset.html#class-tileset-method-get-pattern), [TileSet.remove_pattern](https://docs.godotengine.org/en/4.0/classes/class_tileset.html#class-tileset-method-remove-pattern). So "Index" might make more sense than "ID".